### PR TITLE
Build rootfs filesystems more cleanly with just docker and export to squashfs (avoids requiring root privileges!)

### PR DESCRIPTION
**Pull request by alevy**
This PR replaces the legacy build system for runtime file systems with one based entirely on layered docker containers and then exported to a squashfs file system. Overall this has three main advantages:

1. Building root file systems is faster most of the time
2. The resulting rootfs file systems are smaller
3. Building doesn't require root (because we can make a squashfs file system without root, unlike ext4).

(Requires the `squashfs-tools-ng` package for `gensquashfs`)

### CODE REVIEW

File: rootfs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM  docker.io/library/alpine:3.16
+
+RUN apk add openrc util-linux build-base ca-certificates rng-tools rng-tools-openrc
+RUN rc-update add devfs boot
+RUN rc-update add procfs boot
+RUN rc-update add rngd boot
```
Comment (tan-yue) [inline rootfs/Dockerfile:6]: Is this the elegant solution to the node-hanging-forever problem?

File: rootfs/docker_build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
```
Comment (tan-yue) [inline rootfs/docker_build.sh:2]: Can we add a check and a help message here asking for exactly one positional parameter?

**Review by tan-yue — CHANGES_REQUESTED** [review]

File: rootfs/Dockerfile (hunk 1 above)
Comment (alevy) [reply rootfs/Dockerfile:6]: Yes. It isn't new, it's already in the root image, but rngd collects randomness from the cpu random number generator so fills the randomness pool immediately in most cases.

**Comment by alevy** [comment]
@tan-yue addressed your comment

**Review by tan-yue — APPROVED** [review]

**Pull request merged**